bin/alluxio-monitor.sh

```diff
@@ -175,27 +175,44 @@ run_monitors() {
     nodes=$(echo -e "${nodes}" | tail -n+2)
     if [[ $? -ne 0 ]]; then
       # if there is an error, print the log tail for the remaining master nodes.
-      for node in $(echo ${nodes}); do
-        run_on_node ${node} "${BIN}/alluxio-monitor.sh" -L "${node_type}"
-      done
+      batch_run_on_nodes "$(echo ${nodes})" "${BIN}/alluxio-monitor.sh" "-L ${node_type}"
     else
       HA_ENABLED=$(${BIN}/alluxio getConf ${ALLUXIO_MASTER_JAVA_OPTS} alluxio.zookeeper.enabled)
       JOURNAL_TYPE=$(${BIN}/alluxio getConf ${ALLUXIO_MASTER_JAVA_OPTS} alluxio.master.journal.type | awk '{print toupper($0)}')
       if [[ ${JOURNAL_TYPE} == "EMBEDDED" ]]; then
         HA_ENABLED="true"
       fi
       if [[ ${HA_ENABLED} == "true" ]]; then
-        for node in $(echo ${nodes}); do
-          run_on_node ${node} "${BIN}/alluxio-monitor.sh" ${mode} "${node_type}"
-        done
+        batch_run_on_nodes "$(echo ${nodes})" "${BIN}/alluxio-monitor.sh" "${mode} ${node_type}"
       fi
     fi
   else
-    for node in $(echo "${nodes}"); do
-      run_on_node ${node} "${BIN}/alluxio-monitor.sh" ${mode} "${node_type}"
-    done
+    batch_run_on_nodes "$(echo ${nodes})" "${BIN}/alluxio-monitor.sh" "${mode} ${node_type}"
   fi
+}
+
+# Used to run a command on multiple hosts concurrently.
+# By default it limits concurrent tasks to 100.
+batch_run_on_nodes() {
+  # String of nodes, seperated by a new line
+  local nodes=$1
+  # Command to run on each node
+  local command=$2
+  # Parameter for command
+  local params=$3
```

Comment:
I'm thinking now it might be better to utilize `shift` (http://tldp.org/LDP/Bash-Beginners-Guide/html/sect_09_07.html) and then for params you should be able to use `$@` instead to get the remaining arguments. This way you don't need to quote the final set of args to the function.

Reply:
Alternatively without using `shift` you could do `${@:3}` to append the 3rd and every subsequent arg into a single var here

Reply:
I liked the second one better. Thanks!